neqo-http3/src/connection_client.rs

```diff
@@ -5562,4 +5656,311 @@ mod tests {
             assert_closed(&client, &Error::HttpSettings);
         }
     }
+
+    #[test]
+    fn response_w_1xx() {
+        let (mut client, mut server, request_stream_id) = connect_and_send_request(true);
+
+        setup_server_side_encoder(&mut client, &mut server);
+
+        let headers1xx = vec![(String::from(":status"), String::from("101"))];
+
+        let encoded_headers = server
+            .encoder
+            .encode_header_block(&mut server.conn, &headers1xx, request_stream_id)
+            .unwrap();
+        let hframe = HFrame::Headers {
+            header_block: encoded_headers.to_vec(),
+        };
+        let mut d = Encoder::default();
+        hframe.encode(&mut d);
+        let headers200 = vec![
+            (String::from(":status"), String::from("200")),
+            (String::from("my-header"), String::from("my-header")),
+            (String::from("content-length"), String::from("3")),
+        ];
+
+        let encoded_headers = server
+            .encoder
+            .encode_header_block(&mut server.conn, &headers200, request_stream_id)
+            .unwrap();
+        let hframe = HFrame::Headers {
+            header_block: encoded_headers.to_vec(),
+        };
+        hframe.encode(&mut d);
+        // Send 1xx and 200 headers response.
+        server_send_response_and_exchange_packet(
+            &mut client,
+            &mut server,
+            request_stream_id,
+            &d,
+            false,
+        );
+
+        server_send_response_and_exchange_packet(
+            &mut client,
+            &mut server,
+            request_stream_id,
+            &[0x0, 0x0, 0x0],
```

Comment:
You probably need to explain what this means.  It took me a while to understand it.  Isn't this an empty DATA frame and one more byte?  Why don't we choke on that?

Reply:
This data is not important. I will change it to be something that make sense.
This is legal. it is DATA frame of length 0 and another incomplete DATA frame.